tokio/src/stream/skip_while.rs

```diff
@@ -46,30 +43,30 @@ where
     type Item = St::Item;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        if self.ready {
-            self.as_mut().project().stream.poll_next(cx)
-        } else {
+        if self.predicate.is_some() {
```

Comment:
If you put an `if let Some(predicate) = ...` here, wouldn't you be able to avoid the unwrap below? You will possibly also have to reuse the output of `project()`, e.g. `let this = self.project()` before the if and use that afterwards.

Reply:
Thanks, those are good suggestions. :+1: 
I was thinking about the same, but I didn't manage to get my head around the borrow checker. But I gave it another shot now and it seems a lot better.